src/dataflow-types/src/client/controller/compute.rs

```diff
@@ -91,7 +93,10 @@ impl<T> ComputeControllerState<T>
 where
     T: Timestamp + Lattice,
 {
-    pub(super) fn new(client: Box<dyn ComputeClient<T>>, logging: &Option<LoggingConfig>) -> Self {
+    pub(super) fn new(
+        client: ActiveReplication<Box<dyn ComputeClient<T>>, T>,
+        logging: &Option<LoggingConfig>,
+    ) -> Self {
```

Comment:
Could potentially construct it without a client, and require a call to `add_replica`.

Reply:
I did that, as it appeared to be the easiest way to make things correct (otherwise, things start up in an inconsistent state, and catch on fire).

Reply:
SGTM!